docs/src/modules/components/CodeEditor.js

```diff
@@ -0,0 +1,164 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import Editor from 'react-simple-code-editor';
+import Box from '@mui/material/Box';
+import { styled } from '@mui/material/styles';
+import prism from '@mui/markdown/prism';
+import MarkdownElement from 'docs/src/modules/components/MarkdownElement';
+import CodeCopyButton from 'docs/src/modules/components/CodeCopyButton';
+import { useTranslate } from 'docs/src/modules/utils/i18n';
+import { useCodeCopy } from 'docs/src/modules/utils/CodeCopy';
+import { blue, blueDark } from 'docs/src/modules/brandingTheme';
+
+const Wrapper = styled(MarkdownElement)(({ theme }) => ({
+  position: 'relative',
+  padding: 0,
+  marginBottom: theme.spacing(1),
+  marginTop: theme.spacing(2),
+  [theme.breakpoints.up('sm')]: {
+    marginTop: theme.spacing(0),
+  },
+  borderRadius: theme.shape.borderRadius,
+  '&:hover': {
+    outline: `3px solid ${
+      theme.palette.mode === 'dark' ? theme.palette.primaryDark['500'] : theme.palette.primary.light
+    }`,
+    '& .MuiCode-copy': {
+      opacity: 1,
+    },
+  },
+  '&:focus-within': {
```

Comment:
@mbrookes I'm moving the previous discussion to the code side, so we can automatically resolve it once solved. I don't think that we should keep commenting on the main comment space of the PR. I have resolved all the comments. From what I understood, this is the relevant line of:

> The editor has a keyboard focus indicator when selected by mouse.

from https://github.com/mui/material-ui/pull/32107#issuecomment-1092173585.

But I don't understand the point that you are trying to make, I'm on the same page as @nihgwu. If you could expand on the current vs. expected behavior and why it should behave differently, it would be great.

---

My best assumption is that you are saying this should be:
```suggestion
  '&:focus-visible': {
```

Which would go against:

- How the platform behaves https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input. To be clear, the input on the homepage is not representative of how the platform behaves because it has an `outline: 0` style:

<img width="380" alt="Screenshot 2022-07-18 at 12 16 14" src="https://user-images.githubusercontent.com/3165635/179491261-44bffb5b-5cb1-4e32-b1a8-f09c0ef5c5df.png">

that I think we should fix, e.g. Tailwind does it correct https://tailwindcss.com/blog.

<img width="379" alt="Screenshot 2022-07-18 at 12 12 24" src="https://user-images.githubusercontent.com/3165635/179490666-513d90fc-4998-44e8-a805-1c0d56088d78.png">


- https://youtu.be/ilj2P5-5CjI?t=85 "But for something like a text field, it can be really, really helpful to have that focus indicator. […] And maybe you look away for a bit, and you come back." (meaning when focusing with the mouse)
- how it's implemented https://github.com/WICG/focus-visible/blob/ea5722da398ae6cdc3bb23251486e07985d5e2e4/src/focus-visible.js#L48-L59
- how Ant Design, Material Design, Bootstrap, Tailwind, Fluent UI, etc, behave.